### PR TITLE
Add Windows x64/arm64 GitHub CI workflow to Flang project

### PR DESCRIPTION
**Pull request by kaadam**

### CODE REVIEW

**Comment by kaadam** [comment]
You can check the test workflow in my fork: https://github.com/kaadam/flang/actions/runs/4052923051

File: .github/workflows/build_flang_windows_x64.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          & 7z x "$pwd\llvm-build-win64.7z" -o"$pwd\classic-flang-llvm-project\" -y
+          write-output "$(Get-ChildItem)"
+
+      - name: Build Flang & Libpgmath
+        run: |
+          python .\scripts\build_flang.py -d build -p "$(pwd)\..\..\classic-flang-llvm-project\classic-flang-llvm-project\build\"
+
+      - name: Copy llvm-lit
+        run: |
+          Copy-Item "$(pwd)\..\..\classic-flang-llvm-project\classic-flang-llvm-project\build\" -Destination build\bin\.
```
Comment (bryanpkc) [inline .github/workflows/build_flang_windows_x64.yml:85]: Why isn't there a `Test flang` step after this?

File: .github/workflows/build_flang_windows_x64.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          write-output "$($url)"
+          $artifactPath = "$pwd\llvm_build_win64_clangcl_${{ matrix.llvm_branch }}.zip"
+          Invoke-RestMethod -Method Get -Uri $url -OutFile $artifactPath -Headers @{ "Authorization" = "Bearer ${{ secrets.GITHUB_TOKEN }}" }
+          unzip -o llvm_build_win64_clangcl_${{ matrix.llvm_branch }}.zip
+          & 7z x "$pwd\llvm-build-win64.7z" -o"$pwd\classic-flang-llvm-project\" -y
+          write-output "$(Get-ChildItem)"
+
+      - name: Build Flang & Libpgmath
+        run: |
+          python .\scripts\build_flang.py -d build -p "$(pwd)\..\..\classic-flang-llvm-project\classic-flang-llvm-project\build\"
```
Comment (bryanpkc) [inline .github/workflows/build_flang_windows_x64.yml:81]: Why doesn't this step install flang? The step name should be `Build and install flang & libpgmath` to be consistent with other jobs.

File: .github/workflows/build_flang_windows_x64.yml (hunk 1 above)
Comment (kaadam) [reply .github/workflows/build_flang_windows_x64.yml:85]: Currently we're not able to run Flang tests on Windows, that's why this step is missing from the workflow. Shall I add this step with a proper comment?

File: .github/workflows/build_flang_windows_x64.yml (hunk 2 above)
Comment (kaadam) [reply .github/workflows/build_flang_windows_x64.yml:81]: Basically it does. The build script will install pgmath and flang into '-p "$(pwd)\..\..\classic-flang-llvm-project\classic-flang-llvm-project\build\'.  But yes, a clean directory would be better.

File: .github/workflows/build_flang_windows_x64.yml (hunk 1 above)
Comment (bryanpkc) [reply .github/workflows/build_flang_windows_x64.yml:85]: Do you know what is blocking the X86 Windows build from running Fortran tests?

File: .github/workflows/build_flang_windows_x64.yml (hunk 2 above)
Comment (bryanpkc) [reply .github/workflows/build_flang_windows_x64.yml:81]: I see, the script performs installation unconditionally.

File: .github/workflows/build_flang_windows_x64.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+      - name: Check tools
+        run: |
+          git --version
+          cmake --version
+          clang --version
+          ninja --version
+
+      # Download artifacts for the classic-flang-llvm-project-related builds (all toolchains)
+      - name: Download and extract LLVM artifact
```
[the remark this answers is not in the document]
Comment (kaadam) [reply .github/workflows/build_flang_windows_x64.yml]: Done

File: .github/workflows/build_flang_windows_x64.yml (hunk 2 above)
Comment (kaadam) [reply .github/workflows/build_flang_windows_x64.yml]: Done

File: .github/workflows/build_flang_windows_x64.yml (hunk 1 above)
Comment (kaadam) [reply .github/workflows/build_flang_windows_x64.yml]: The current Flang tests results on Windows x64:
Testing Time: 15.74sUnsupported      :  369                                                                                                 Passed           :   54                                                                                                 Expectedly Failed:    1                                                                                                 Failed           : 1397

Comment (bryanpkc) [reply .github/workflows/build_flang_windows_x64.yml]: Yeah, the legacy test suite (`f90_correct` etc.) is not portable and incurs a lot of unnecessary duplication. I have always wished we could revamp the system.

**Review by bryanpkc — APPROVED** [review]
LGTM

**Comment by bryanpkc** [comment]
Merging CI changes with one approval.

**Pull request merged**